lib/model_renamer/rename.rb

```diff
@@ -71,17 +76,18 @@ def replace_in_file filepath, old_string, new_string
     end
   end
 
-  def rename_file filepath
-    variation_pairs.each do |variation|
-      create_directory filepath, variation[0], variation[1]
-      File.rename(filepath, filepath.gsub(variation[0], variation[1])) if File.file?(filepath)
+  def rename_files
+    all_filepaths.product(variation_pairs).each do |filepath, (old_name, new_name)|
+      next unless File.basename(filepath).include? old_name
+      filename = File.basename(filepath)
+      File.rename filepath, filepath.gsub(filename, filename.gsub(old_name, new_name))
```

Comment:
Cant he second argument be extracted out as something like `new_filepath`, to keep this lines easier to read? rename with gsub and a nested gsub make it hard to grok the first readthrough.